srcpkgs/OpenJK/template

```diff
@@ -0,0 +1,47 @@
+# Template file for 'OpenJK'
+pkgname=OpenJK
+version=0.0.20210922
+revision=1
+archs="i686 x86_64"
+_gitrev=43e9a3da555a487e23081bbd03101e7c926a1e83
+wrksrc="${pkgname}-${_gitrev}"
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/opt/${pkgname} -DBuildJK2SPEngine=on -DBuildJK2SPGame=on -DBuildJK2SPRdVanilla=on"
+makedepends="libjpeg-turbo-devel libpng-devel zlib-devel SDL2-devel"
+depends="SDL2"
+short_desc="Open Source Implementation of Jedi Knight II and III Engine"
+maintainer="Kenneth Dodrill <kmdodrill@protonmail.com>"
+license="GPL-2.0-only"
+homepage="https://github.com/JACoders/OpenJK"
+distfiles="https://github.com/JACoders/OpenJK/archive/${_gitrev}.tar.gz"
+checksum=efd2e3d96fe449819c4276dfea7ae7d591591ee0e651167844110e073f2edb3c
+
+post_install() {
+	vmkdir usr/bin
+	for _bin in 'openjk' 'openjk_sp' 'openjkded'; do
+		touch "${DESTDIR}/usr/bin/${_bin}"
+		cat > "${DESTDIR}/usr/bin/${_bin}" << EOF
+#! /bin/bash
+cd /opt/${pkgname}/JediAcademy
+exec ./${_bin}.${XBPS_MACHINE} \s@
```

Comment:
`\s@` should be `\$@`?

Reply:
Add this in do_install and replace your occurrences of XBPS_MACHINE with arch, this is mainly what we set in the cmake build-style, but i686 changed to i366 instead of x86
```
case $XBPS_TARGET_ARCH in
	arm*) arch=arm;;
	aarch64*) arch=aarch64;;
	i686*) arch=x86;;
	mips*) arch=mips;;
	x86_64*) arch=x86_64;;
	ppc64le*) arch=ppc64le;;
	ppc64*) arch=ppc64;;
	ppcle*) arch=ppcle
	ppc*) arch=ppc
	*) arch="generic"
esac
```

Reply:
no, they should just do what i said there (ask for `uname -m` at runtime)

Reply:
Or the shorter version as by q66 comment,
```
case $XBPS_TARGET_ARCH in
	i686*) arch=i386
	*) arch=$(uname -m)
esac
```

Reply:
> no, they should just do what i said there (ask for `uname -m` at runtime)

The cmake stuff overwrites some names, but for us only the i686 -> i386 is important.

Reply:
actually looks like more might be important

so the script should do something like

```
OJ_ARCH=$(uname -m)
case "$OJ_ARCH" in
    arm*) OJ_ARCH=arm;;
    i*86*) OJ_ARCH=i386;;
esac
```